bt-core/src/main/java/bt/torrent/messaging/TorrentWorker.java

```diff
@@ -105,6 +110,47 @@ public TorrentWorker(TorrentId torrentId,
         eventSource.onPeerConnected(torrentId, e -> onPeerConnected(e.getConnectionKey()));
 
         eventSource.onPeerDisconnected(torrentId, e -> onPeerDisconnected(e.getConnectionKey()));
+
+        startAssignmentsUpdateThread(torrentId, eventSource, config);
+    }
+
+    private void startAssignmentsUpdateThread(TorrentId torrentId, EventSource eventSource, Config config) {
```

Comment:
Same as for DataWorker, I'm concerned about potential concurrency issues (with regards to the objects that are updated here and then used in the other parts of the program).